File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/C/s504715611_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

typedef struct {
    int *parent;
    int size;
} UnionFind;

UnionFind* createUnionFind(int n);
int root(UnionFind* uf, int A);
int sizeUnion(UnionFind* uf, int A);
int connectUnion(UnionFind* uf, int A, int B);

int lower_bound(int a[], int n, int key);
int upper_bound(int a[], int n, int key);
int isPrime(long long n);
int upper_division(int a, int b);
long long lupper_division(long long a, long long b);
int* setArray(int a);
long long* lsetArray(int a);
char* reverse(char* str);
void printArray(int que[], int len);
int compareIntArrays(const void *a, const void *b);
void doublesort(int **a, int n);
int compareLongLongArrays(const void *a, const void *b);
void ldoublesort(long long **a, int n);
long long modpow(long long x, long long n, long long mo);
char* revch(char* ch);
int* revint(int ch[], int len);
void warshall_floyd(int **v, int n);
long long modint(long long a, long long m);

UnionFind* createUnionFind(int n) {
    UnionFind* uf = malloc(sizeof(UnionFind));
    uf->parent = malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) uf->parent[i] = -1;
    uf->size = n;
    return uf;
}

int root(UnionFind* uf, int A) {
    if (uf->parent[A] < 0) return A;
    return uf->parent[A] = root(uf, uf->parent[A]);
}

int sizeUnion(UnionFind* uf, int A) {
    return -uf->parent[root(uf, A)];
}

int connectUnion(UnionFind* uf, int A, int B) {
    A = root(uf, A);
    B = root(uf, B);
    if (A == B) return 0;
    if (sizeUnion(uf, A) < sizeUnion(uf, B)) {
        int temp = A;
        A = B;
        B = temp;
    }
    uf->parent[A] += uf->parent[B];
    uf->parent[B] = A;
    return 1;
}

int lower_bound(int a[], int n, int key) {
    int right = n;
    int left = 0;
    while (right - left > 1) {
        int mid = (right + left) / 2;
        if (a[mid] < key) left = mid;
        else right = mid;
    }
    return left;
}

int upper_bound(int a[], int n, int key) {
    int right = n;
    int left = 0;
    while (right - left > 1) {
        int mid = (right + left) / 2;
        if (a[mid] <= key) left = mid;
        else right = mid;
    }
    return left;
}

int isPrime(long long n) {
    if (n == 2) return 1;
    if (n < 2 || n % 2 == 0) return 0;
    long long d = sqrt(n);
    for (long long i = 3; i <= d; i += 2) {
        if (n % i == 0) return 0;
    }
    return 1;
}

int upper_division(int a, int b) {
    return (a % b == 0) ? (a / b) : (a / b + 1);
}

long long lupper_division(long long a, long long b) {
    return (a % b == 0) ? (a / b) : (a / b + 1);
}

int* setArray(int a) {
    int *b = malloc(a * sizeof(int));
    for (int i = 0; i < a; i++) {
        scanf("%d", &b[i]);
    }
    return b;
}

long long* lsetArray(int a) {
    long long *b = malloc(a * sizeof(long long));
    for (int i = 0; i < a; i++) {
        scanf("%lld", &b[i]);
    }
    return b;
}

char* reverse(char* str) {
    int len = strlen(str);
    char *ch = malloc(len + 1);
    char *chch = str;
    int a = len;
    for (int i = 0; i < upper_division(a, 2); i++) {
        ch[i] = chch[len - i - 1];
        ch[len - 1 - i] = chch[i];
    }
    ch[len] = '\0';
    return ch;
}

void printArray(int que[], int len) {
    for (int i = 0; i < len - 1; i++) {
        printf("%d ", que[i]);
    }
    if (len > 0) printf("%d\n", que[len - 1]);
}

int compareIntArrays(const void *a, const void *b) {
    int **x = (int **)a;
    int **y = (int **)b;
    return (*x)[0] - (*y)[0];
}

void doublesort(int **a, int n) {
    qsort(a, n, sizeof(int *), compareIntArrays);
}

int compareLongLongArrays(const void *a, const void *b) {
    long long **x = (long long **)a;
    long long **y = (long long **)b;
    return (*x)[0] - (*y)[0];
}

void ldoublesort(long long **a, int n) {
    qsort(a, n, sizeof(long long *), compareLongLongArrays);
}

long long modpow(long long x, long long n, long long mo) {
    long long sum = 1;
    while (n > 0) {
        if (n & 1) {
            sum = (sum * x) % mo;
        }
        x = (x * x) % mo;
        n >>= 1;
    }
    return sum;
}

char* revch(char* ch) {
    int len = strlen(ch);
    char* ret = malloc(len + 1);
    for (int i = len - 1, j = 0; i >= 0; i--, j++) {
        ret[j] = ch[i];
    }
    ret[len] = '\0';
    return ret;
}

int* revint(int ch[], int len) {
    int* ret = malloc(len * sizeof(int));
    for (int i = len - 1, j = 0; i >= 0; i--, j++) {
        ret[j] = ch[i];
    }
    return ret;
}

void warshall_floyd(int **v, int n) {
    for (int k = 0; k < n; k++) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                v[i][j] = (v[i][j] < (v[i][k] + v[k][j])) ? v[i][j] : (v[i][k] + v[k][j]);
            }
        }
    }
}

long long modint(long long a, long long m) {
    long long b = m, u = 1, v = 0;
    while (b != 0) {
        long long t = a / b;
        a -= t * b;
        long long x = a;
        a = b;
        b = x;

        u -= t * v;
        x = u;
        u = v;
        v = x;
    }
    u %= m;
    if (u < 0) u += m;
    return u;
}

int main() {
    char str[10];
    if (scanf("%9s", str) != 1) return 1;
    int len = strlen(str);
    if (len < 6) {
        printf("No\n");
    } else {
        if (str[2] == str[3] && str[4] == str[5]) {
            printf("Yes\n");
        } else {
            printf("No\n");
        }
    }
    return 0;
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s504715611_commented.py
mod = 10**9 +7

class UnionFind:
    def __init__(self, n):
        self.Parent = [-1]*n
    def root(self, A):
        if self.Parent[A] <0:
            return A
        self.Parent[A] = self.root(self.Parent[A])
        return self.Parent[A]
    def size(self, A):
        return -self.Parent[self.root(A)]
    def connect(self, A, B):
        A_root = self.root(A)
        B_root = self.root(B)
        if A_root == B_root:
            return False
        if self.size(A) < self.size(B):
            A_root, B_root = B_root, A_root
        self.Parent[A_root] += self.Parent[B_root]
        self.Parent[B_root] = A_root
        return True

def gcd(a, b):
    while b !=0:
        a, b = b, a%b
    return a

def modint(a, m):
    b = m
    u = 1
    v = 0
    while b !=0:
        t = a // b
        rem = a - t * b
        a, b = b, rem
        temp_u = u - t * v
        u, v = v, temp_u
    u %= m
    if u <0:
        u += m
    return u

def modlcm(a, b):
    g = gcd(a, b)
    return (a * b * modint(g, mod)) % mod

def lcm(a, b):
    return a * b // gcd(a, b)

def max(a, b):
    return a if a > b else b

def min(a, b):
    return a if a < b else b

def lmax(a, b):
    return max(a, b)

def lmin(a, b):
    return min(a, b)

def factorial(i):
    if i == 1:
        return 1
    return i * factorial(i-1)

def lower_bound(a, key):
    right = len(a)
    left =0
    while right - left >1:
        mid = (right + left)//2
        if a[mid] < key:
            left = mid
        else:
            right = mid
    return left

def upper_bound(a, key):
    right = len(a)
    left =0
    while right - left >1:
        mid = (right + left)//2
        if a[mid] <= key:
            left = mid
        else:
            right = mid
    return left

import math

def isPrime(n):
    if n ==2:
        return True
    if n <2 or n%2 ==0:
        return False
    d = math.sqrt(n)
    for i in range(3, int(d)+1, 2):
        if n%i ==0:
            return False
    return True

def upper_division(a, b):
    return (a + b -1) // b

def lupper_division(a, b):
    return (a + b -1) // b

def setArray(a):
    res = [0]*a
    for i in range(a):
        res[i] = scan.nextInt()
    return res

def lsetArray(a):
    res = [0]*a
    for i in range(a):
        res[i] = scan.nextLong()
    return res

def reverse(s):
    ch = list(s)
    a = len(ch)
    for i in range( (a +1)//2 ):
        ch[i], ch[a-1 -i] = ch[a-1 -i], ch[i]
    return ''.join(ch)

def printArray(arr):
    for i in range(len(arr)-1):
        print(arr[i], end=' ')
    print(arr[-1])

def doublesort(a):
    a.sort(key=lambda x: x[0])
    return a

def ldoublesort(a):
    a.sort(key=lambda x: x[0])
    return a

def modpow(x, n, mo):
    res = 1
    while n >0:
        if n &1:
            res = res * x % mo
        x = x * x % mo
        n >>=1
    return res

def revch(ch):
    return ch[::-1]

def revint(ch):
    return ch[::-1]

def warshall_floyd(v, n):
    for k in range(n):
        for i in range(n):
            for j in range(n):
                v[i][j] = min(v[i][j], v[i][k] + v[k][j])

class FastScanner:
    def __init__(self):
        import sys
        self.tokens = []
        self.ptr = 0
        self.buf = sys.stdin.read().split()
    def next(self):
        if self.ptr >= len(self.buf):
            raise Exception("No more tokens")
        res = self.buf[self.ptr]
        self.ptr +=1
        return res
    def nextInt(self):
        return int(self.next())
    def nextLong(self):
        return int(self.next())
    def nextDouble(self):
        return float(self.next())

def main():
    global scan
    scan = FastScanner()
    str = scan.next()
    if str[2] == str[3] and str[4] == str[5]:
        print("Yes")
    else:
        print("No")

if __name__ == "__main__":
    main()  # End of Code
